src/Hosting.CommandLine/ICliService.cs

```diff
@@ -0,0 +1,19 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace McMaster.Extensions.Hosting.CommandLine
+{
+    /// <summary>
+    /// A service to be run as part of the <see cref="CliLifetime"/>.
+    /// </summary>
+    /// <seealso cref="HostBuilderExtensions.RunCliAsync{T}(Microsoft.Extensions.Hosting.IHostBuilder, string[])"/>
+    public interface ICliService
```

Comment:
Do you envision someone implementing this on their own? I'm wondering if this could be internal as well.

Reply:
I wasn't sure about this.  Maybe I don't fully understand `internal`, but if I mark the interface internal, i get:
```
CliLifetime.cs(22,16): error CS0051: Inconsistent accessibility: parameter type 'ICliService' is less accessible than method 'CliLifetime.CliLifetime(ILogger<CliLifetime>, IApplicationLifetime, ICliService)' [C:\Users\lucas\git\pastdev-commandlineutils\src\Hosting.CommandLine\McMaster.Extensions.Hosting.CommandLine.csproj]
```
Suggestion?

Reply:
`internal` is viral. When you make `ICliService` internal, all other API that uses it must also be `internal`, so `CommandLineLifetime` would also need to be internal.

My suggestion is make most things `internal`, only making API public when it's clear that someone would use the API. I've learned the hard way that once you make an API public, you can never change it, so I'm really conservative on this one. It gives us flexibility to refactor and change types, interfaces, names, etc. later without breaking people.

Reply:
I totally agree, keep the implementation details internal.  Just got my wires crossed on the constructor allowing for the internal `ICliService`...  All [fixed now](https://github.com/natemcmaster/CommandLineUtils/pull/167/commits/fcbb8afd322432bd6ff76f33c2c1abd5832726f6)

_I made everything except the extension methods internal_